redis.conf

```diff
@@ -70,6 +70,29 @@ tcp-backlog 511
 # unixsocket /tmp/redis.sock
 # unixsocketperm 700
 
+# Enable forcing of TCP connections over SSL. 
+# If enabled, you must specify either a certificate root store file or directory (not both) in 
+# PEM format (or a set of PEM files if a directory). You must also specify the certificate for the
+# server, the private key file, and the Diffie-Hellman parameters file. These may be in separate files
+# or in the same file. All files must be in PEM format (as required by openSSL). 
+# If you wish, you can customize the cipher list to use when accepting SSL connections. By default, the list
+# By default, we use all ciphers with the exception of: ones with anonymous Diffie-Hellman, low-strength ciphers, 
+# export ciphers, md5 hashing. Ordered from strongest to weakest.
+# Note that as ciphers become broken, it will be necessary to change the available cipher list to remain secure. 
+#
+# If you are using slave processes, and wish to validate the common name of the redis server, you can
+# specify the parameter ssl_cert_common_name with the name you wish to validate against.
+#
+# ssl true
+# ssl_ca_root_dir < Path to your openSSL root certificates directory >
+# ssl_ca_root_file < Path to your openSSL root certificate file (You must specify either a root CA file, or a directory, but not both) >
+# ssl_cert_file <Path to your server's SSL cert>
+# ssl_pk_file <Path to your server's private key file (may be the same as above)>
+# ssl_dhk_file <Path to your Diffie-Hellman data file (may be the same as above)>
+# ssl_cert_pass <optional - password for server cert file>
+# ssl_cipher_list <optional - cipher list to use for the server. Defaults to ALL:!ADH:!LOW:!EXP:!MD5:@STRENGTH>
+# ssl_cert_common_name <optional - common name of server cert to match>
+
```

Comment:
I'm mostly familiar with Ruby's OpenSSL bindings, where their SSLContext object has a `ssl_version` setting: https://github.com/basho/riak-ruby-client/blob/master/lib/riak/client/beefcake/socket.rb#L72-L87

Looking at http://www.openssl.org/docs/ssl/SSL_CTX_new.html , I suspect this is what Ruby's OpenSSL does:
> SSLv23_method(void), SSLv23_server_method(void), SSLv23_client_method(void)
> A TLS/SSL connection established with these methods may understand the SSLv3, TLSv1, TLSv1.1 and TLSv1.2 protocols.
>
> If extensions are required (for example server name) a client will send out TLSv1 client hello messages including extensions and will indicate that it also understands TLSv1.1, TLSv1.2 and permits a fallback to SSLv3. A server will support SSLv3, TLSv1, TLSv1.1 and TLSv1.2 protocols. This is the best choice when compatibility is a concern.

Mandatory security activism question: does anybody need a "secure" Redis client in an environment where TLSv1 or later isn't available? Is there a benefit to supporting the seriously and unfixably broken SSLv3?